### PR TITLE
Update instance segmentation

### PR DESCRIPTION
**Pull request by DmitriySidnev**
These are new instance segmentation models.

### CODE REVIEW

**Comment by eaidova** [comment]
Jenkins ok to test

File: models/intel/instance-segmentation-security-0091/accuracy-check.yml
Comment (eaidova) [inline models/intel/instance-segmentation-security-0091/accuracy-check.yml:25]:
```suggestion
            use_numpy: True
```

File: tools/accuracy_checker/accuracy_checker/adapters/mask_rcnn.py
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/mask_rcnn.py:250]: fast fix for linter issue
```suggestion
                    per_obj_raw_masks.append(raw_mask[cls, ...] if self.scores_out else raw_mask)
```

**Comment by eaidova** [comment]
@DmitriySidnev please add new models to index.md

File: models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md
Comment (druzhkov-paul) [inline models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md:19]: This 1000 does not match with shape of network outputs described below.

Comment (DmitriySidnev) [reply models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md]: Fixed to the output shape.

File: models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md
Comment (eizamaliev) [inline models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md:36]: May be `[100,1]` or `[100]`?

Comment (DmitriySidnev) [reply models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md]: Fixed to `[100]`.

File: models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md
Comment (vladimir-dudnik) [inline models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md:27]: we are removing empty sections from documentation (see [PR-1749](https://github.com/openvinotoolkit/open_model_zoo/pull/1749)), please do this as well

**Review by vladimir-dudnik — CHANGES_REQUESTED** [review]

File: models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md
Comment (DmitriySidnev) [reply models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md]: Fixed.

**Review by vladimir-dudnik — APPROVED** [review]

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
please add soft link to AC config from old configs location

**Review by vladimir-dudnik — CHANGES_REQUESTED** [review]
do not merge until accuracy validation release criteria met

**Comment by DmitriySidnev** [comment]
@vladimir-dudnik, @eaidova, should I do something else except accuracy-checker results? CI failed for Demos because demos ran with old models that are not compatible with new ones and Documentation because in `index.md` are not present old models now.

**Comment by vladimir-dudnik** [comment]
@DmitriySidnev we will return to this PR once new models merged. This will not go for 2021.2 release.

**Comment by DmitriySidnev** [comment]
@vladimir-dudnik, I have updated metrics. For two models 0002 and 0091 that can't be preprocessed as in Pytorch I added metrics for both options as it was early for old models. For other models also were slightly reduced metrics. Please take a look.

**Review by vladimir-dudnik — APPROVED** [review]
model now fits release criteria

**Comment by vladimir-dudnik** [comment]
@DmitriySidnev in docs there are link to replaced models:
17:57:51  [✖] ./instance-segmentation-security-0904/description/instance-segmentation-security-0904.md
17:57:51  [✖] ./instance-segmentation-security-0912/description/instance-segmentation-security-0912.md

**Comment by vladimir-dudnik** [comment]
@DmitriySidnev now the issue is that:
13:10:23  instance-segmentation-security-0050 not in index.md file

File: tools/accuracy_checker/accuracy_checker/adapters/mask_rcnn.py
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/mask_rcnn.py:183]: not sure that it is necessary, we prefer to align classes in annotation with model labels, there is option which allow convert dataset without background label

Comment (DmitriySidnev) [reply tools/accuracy_checker/accuracy_checker/adapters/mask_rcnn.py]: Fixed.

**Comment by vladimir-dudnik** [comment]
14:18:50  demos/instance_segmentation_demo/python/instance_segmentation_demo/model_utils.py:239: trailing whitespace.
14:18:50  +    'mask_rcnn_segmentoly': ModelAttributes(('boxes', 'scores', 'classes', 'raw_masks'), 
14:18:50  running yamllint...
14:18:56  running flake8...
14:19:04  ./demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py:37:16: E713 test for membership should be 'not in'
14:19:04  ./demos/multi_camera_multi_target_tracking/python/utils/network_wrappers.py:149:16: E713 test for membership should be 'not in'

File: models/intel/index.md
Comment (IRDonch) [inline models/intel/index.md:147]: Please list the models in alphabetic (well, numeric, in this case) order.

File: models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md
Comment (IRDonch) [inline models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md:27]:
```suggestion
[MS COCO evaluation procedure](https://cocodataset.org/#detection-eval).
```

**Review by IRDonch — CHANGES_REQUESTED** [review]

**Comment by Wovchena** [comment]
Please, extend tests with new models: https://github.com/openvinotoolkit/open_model_zoo/blob/29a8c8cef1cc60496ee04cbf05f8b8974893d238/demos/tests/cases.py#L481

File: demos/instance_segmentation_demo/python/instance_segmentation_demo/model_utils.py
Comment (DmitriySidnev) [reply demos/instance_segmentation_demo/python/instance_segmentation_demo/model_utils.py]: Fixed.

File: demos/multi_camera_multi_target_tracking/python/utils/network_wrappers.py
Comment (DmitriySidnev) [reply demos/multi_camera_multi_target_tracking/python/utils/network_wrappers.py]: Yes, removed.

File: demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py
Comment (DmitriySidnev) [reply demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py]: Fixed.

File: demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py
Comment (DmitriySidnev) [reply demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py]: Fixed.

File: demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py
Comment (DmitriySidnev) [reply demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py:79]: Reimplemented method `segmentoly_type()` with checking all inputs and outputs.

File: models/intel/index.md
Comment (DmitriySidnev) [reply models/intel/index.md]: Fixed.

File: models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md
Comment (DmitriySidnev) [reply models/intel/instance-segmentation-security-0002/description/instance-segmentation-security-0002.md]: Fixed documentation for all new models.

File: demos/multi_camera_multi_target_tracking/python/utils/network_wrappers.py
Comment (IRDonch) [reply demos/multi_camera_multi_target_tracking/python/utils/network_wrappers.py]: There's also similar code in `demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py`.

**Comment by DmitriySidnev** [comment]
```
15:05:56  + ./venv/bin/python ./src/demos/tests/run_tests.py --demo-build-dir=build/intel64/Release --test-data-dir=/mnt/datasets --downloader-cache-dir=/mnt/md-cache --mo=/home/jenkins/workspace/omz/precommit/demos_ubuntu18/openvino/deployment_tools/model_optimizer/mo.py --devices=CPU
15:05:57  Retrieving models...
15:05:59  No matching models: "instance-segmentation-security-0228"
```
Is this error caused by me?

File: demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py
Comment (IRDonch) [reply demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py:79]: It's not quite sufficient, because your `__init__` method still admits models with segmentoly inputs and non-segmentoly outputs, and vice versa. Such models will be treated as non-segmentoly, and either input or output processing will fail.

File: demos/multi_camera_multi_target_tracking_demo/python/utils/network_wrappers.py
Comment (IRDonch) [inline demos/multi_camera_multi_target_tracking_demo/python/utils/network_wrappers.py:151]:
```suggestion
        required_outputs = {'boxes', 'scores', 'classes', 'raw_masks'}
```

File: demos/multi_camera_multi_target_tracking/python/utils/network_wrappers.py
Comment (DmitriySidnev) [reply demos/multi_camera_multi_target_tracking/python/utils/network_wrappers.py]: Fixed.

File: demos/multi_camera_multi_target_tracking_demo/python/utils/network_wrappers.py
Comment (DmitriySidnev) [reply demos/multi_camera_multi_target_tracking_demo/python/utils/network_wrappers.py]: Thanks, fixed.

File: demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py
Comment (DmitriySidnev) [reply demos/whiteboard_inpainting_demo/python/utils/network_wrappers.py:84]: Reimplemented.

**Comment by IRDonch** [comment]
> Is this error caused by me?

The new test cases can't run, because the new models aren't published yet. Comment them out, and we'll reenable them once we publish the models.

Also, note that the style check in the trigger build failed. Please fix the issues that were found.

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

File: demos/multi_camera_multi_target_tracking_demo/python/utils/network_wrappers.py
Comment (Wovchena) [inline demos/multi_camera_multi_target_tracking_demo/python/utils/network_wrappers.py:139]: These two sets are equal. The same is for `check_segmentoly_type()` and whiteboard_inpainting_demo.

**Review by Wovchena — CHANGES_REQUESTED** [review]
`py ..\instance_segmentation_demo\python\instance_segmentation_demo.py -m C:\Users\vzlobin\Downloads\instance-segmentation-security-1039.xml --labels C:\Users\vzlobin\a\z\data\dataset_classes\coco.txt -i 0`
failed with
`While validating node 'v6::ExperimentalDetectronPriorGridGenerator ExperimentalDetectronPriorGridGenerator_1766 (1069/Output_0/Data__const[0]:f16{3,4}, Conv_507[0]:f16{?,3,120,120}, 1070/Output_0/Data__const[0]:f16{1,1,1,1}) -> ()' with friendly_name 'ExperimentalDetectronPriorGridGenerator_1766':
The first dimension of both 'feature_map' and 'im_data' must match. Feature_map: ?; Im_data: 1`
I haven't aligned versions of MO and IE used yet, but may be you have a better guess why it failed.

File: demos/multi_camera_multi_target_tracking_demo/python/utils/network_wrappers.py
Comment (DmitriySidnev) [reply demos/multi_camera_multi_target_tracking_demo/python/utils/network_wrappers.py]: Fixed.

**Comment by DmitriySidnev** [comment]
`I haven't aligned versions of MO and IE used yet, but may be you have a better guess why it failed.`
@Wovchena, I have the only one suggestion - use IE with a model obtained via consistent version of MO.

**Comment by Wovchena** [comment]
It still fails for me. I took IE and MO from 80781ff359f0b7bfd657cd3a78911c4ec108931b commit. Here is what I do
`v` folder stands for cloned https://github.com/openvinotoolkit/openvino, `z` is for https://github.com/openvinotoolkit/open_model_zoo
```PS1
cd a/v/build
cmake -G "Visual Studio 16 2019" -DENABLE_CPPLINT=OFF -DENABLE_CLDNN=OFF -DENABLE_HDDL=OFF -DENABLE_MYRIAD=OFF -DENABLE_TESTS=OFF -DENABLE_BEH_TESTS=OFF -DENABLE_FUNCTIONAL_TESTS=OFF -DENABLE_DLIA=OFF -DENABLE_PRIVATE_MODELS=OFF -DVERBOSE_BUILD=OFF -DENABLE_PERFORMANCE_TESTS=OFF -DENABLE_INTEGRATION_TESTS=OFF -DENABLE_PLUGIN_RPATH=ON -DENABLE_OPENVX_CVE=OFF -DOpenCV_DIR="C:\Program Files (x86)\IntelSWTools\openvino_2021\opencv\cmake" -DENABLE_ICV_TESTS=OFF -DENABLE_MODELS=OFF -DENABLE_OBJECT_DETECTION_TESTS=OFF -DENABLE_SEGMENTATION_TESTS=OFF -DENABLE_VALIDATION_SET=OFF -DENABLE_VPU=OFF -DENABLE_GNA=OFF -DENABLE_PROFILING_ITT=OFF -DTHREADING=TBB -DCMAKE_BUILD_TYPE=Release -DENABLE_PYTHON=ON .. ; cmake --build . -j 4 --config Release
cd ../../z/demos/build
py C:\Users\vzlobin\a\v\model-optimizer\mo_onnx.py --input_model C:\Users\vzlobin\Downloads\model.onnx --reverse_input_channels --mean_values "[ 103.53, 116.28, 123.675 ]" --output boxes,labels,masks --input_shape "[ 1, 3, 480, 480 ]"
$env:PATH="C:\Users\vzlobin\a\v\bin\intel64\Release;C:\Program Files (x86)\IntelSWTools\openvino_2021\opencv\bin;C:\Users\vzlobin\a\v\inference-engine\temp\tbb\bin;$env:PATH"; $env:PYTHONPATH="C:\Users\vzlobin\a\z\demos\build\intel64\Release;C:\Users\vzlobin\a\v\bin\intel64\Release\python_api\python3.7;C:\Program Files (x86)\IntelSWTools\openvino_2021\python\python3;$env:PYTHONPATH"
py ..\instance_segmentation_demo\python\instance_segmentation_demo.py -m C:\Users\vzlobin\a\z\demos\build\.\model.xml --labels C:\Users\vzlobin\a\z\data\dataset_classes\coco.txt -i 0
```
```
[ INFO ] Creating Inference Engine...
[ INFO ] Loading network
Traceback (most recent call last):
  File "..\instance_segmentation_demo\python\instance_segmentation_demo.py", line 239, in <module>
    sys.exit(main() or 0)
  File "..\instance_segmentation_demo\python\instance_segmentation_demo.py", line 110, in main
    net = ie.read_network(args.model, args.model.with_suffix('.bin'))
  File "ie_api.pyx", line 266, in openvino.inference_engine.ie_api.IECore.read_network
  File "ie_api.pyx", line 288, in openvino.inference_engine.ie_api.IECore.read_network
RuntimeError: Check 'num_batches_featmap.same_scheme(num_batches_im_data)' failed at C:\Users\vzlobin\a\v\ngraph\core\src\op\experimental_detectron_prior_grid_generator.cpp:104:
While validating node 'v6::ExperimentalDetectronPriorGridGenerator ExperimentalDetectronPriorGridGenerator_1766 (1069/Output_0/Data__const[0]:f32{3,4}, Conv_507[0]:f32{?,3,120,120}, 1070/Output_0/Data__const[0]:f32{1,1,1,1}) -> ()' with friendly_name 'ExperimentalDetectronPriorGridGenerator_1766':
The first dimension of both 'feature_map' and 'im_data' must match. Feature_map: ?; Im_data: 1
```

I also noticed that it works for `instance-segmentation-security-0228`. May be there is something specific about `instance-segmentation-security-1039`.

**Review by IRDonch — APPROVED** [review]

**Review by eaidova — APPROVED** [review]

**Comment by eaidova** [comment]
> The problem is still there, but it's not about demo.

This PR merging unblocks models validation, so we can see if issue is reproducible on the latest OV

**Pull request merged**
